tests/lib/test_requests.py

```diff
@@ -0,0 +1,237 @@
+import dataclasses
+import json
+import typing
+
+import requests_mock
+
+from fixpoint_sdk.lib import requests as fixrequests
+
+
+@dataclasses.dataclass
+class CallRecords:
+    url: str
+    request: typing.Any
+    response: typing.Any
+
+
+class CallbackRequester(fixrequests.Requester):
+    calls: typing.List[CallRecords]
+
+    def __init__(
+        self,
+        api_key: str,
+        base_url: typing.Optional[str],
+    ):
+        def callback(url, request, response):
+            self.calls.append(CallRecords(url, request, response))
+
+        super().__init__(api_key, base_url, _on_api_call=callback)
+        self.calls = []
+
+
+def new_requester():
+    return CallbackRequester('api_key', 'http://localhost:8081')
+
+
+def test_create_openai_input_log():
+    url, req = new_create_openai_input_log_post_request()
+    out = new_create_openai_input_log_output()
+    args = new_create_openai_input_log_args()
+    requester = new_requester()
+
+    with requests_mock.Mocker() as m:
+        m.post(url, json=out)
+        requester.create_openai_input_log(*args)
+
+    assert requester.calls[0].url == url
+    assert requester.calls[0].request == req
+    assert requester.calls[0].response == out
+
+
+def new_create_openai_input_log_args():
+    model_name = "gpt-3.5-turbo-0125"
```

Comment:
Nit: we may wanna put this value into a constants file

Reply:
This is some test gibberish that I copy pasted by logging outputs. I didn't build these by hand, so I don't want to spend extra time setting up constants here.